### PR TITLE
bug: bump protobuf module to avoid CVE-2024-24786

### PR DESCRIPTION
**Pull request by glitchcrab**
I noticed that builds were failing due to a nancy warning about [CVE-2024-24786](https://app.circleci.com/pipelines/github/giantswarm/deletion-blocker-operator/404/workflows/98c988c2-eb93-4050-b2ba-2e5addafa4db/jobs/962/parallel-runs/0/steps/0-114). Bumping the module seems to work ok on my laptop - the binary builds and runs.

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -54,7 +54,7 @@ require (
 	golang.org/x/time v0.3.0 // indirect
 	gomodules.xyz/jsonpatch/v2 v2.3.0 // indirect
 	google.golang.org/appengine v1.6.7 // indirect
-	google.golang.org/protobuf v1.30.0 // indirect
+	google.golang.org/protobuf v1.33.0 // indirect
```
Comment (erkanerol) [inline go.mod:57]: We should override indirect dependencies via `replace`. This can be overridden by the direct dependency.

Comment (erkanerol) [reply go.mod:57]: e.g. https://github.com/giantswarm/cluster-api-cleaner-vsphere/pull/99

**Comment by glitchcrab** [comment]
@erkanerol i guess i just don't understand the intricacies of managing gomodules well enough. 

for example, i don't understand why the module is [being replaced with the version from github which has a completely different version](https://github.com/giantswarm/cluster-api-cleaner-vsphere/pull/99/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R79)

**Comment by erkanerol** [comment]
> @erkanerol i guess i just don't understand the intricacies of managing gomodules well enough.
> 
> for example, i don't understand why the module is [being replaced with the version from github which has a completely different version](https://github.com/giantswarm/cluster-api-cleaner-vsphere/pull/99/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R79)

We have direct dependencies like `client-go` or `sigs.k8s.io/controller-runtime`. See the first `require` section. Those bring the indirect dependencies and `github.com/golang/protobuf`  is one of them. That module depends on `google.golang.org/protobuf` See https://github.com/golang/protobuf/blob/master/go.mod 

My understanding is that `github.com/golang/protobuf` is spec and `google.golang.org/protobuf` is the implementation. 

We have 2 options here:
1) Update the direct dependencies (client-go or controller-runtime) to bump the indirect dependency -> We need to check if they consume the version we need.
2) Use `replace` to override dependency versions forcefully. 

I used the second one to not touch k8s clients which is more risky.

**Review by erkanerol — APPROVED** [review]

**Pull request merged**